ProjectMacros.cmake

```diff
@@ -112,6 +112,13 @@ macro(CREATE_TEST_TARGETS BASE_NAME SRC DEPENDENCIES)
   if(BUILD_TESTING)
     add_executable(${BASE_NAME}_tests ${SRC})
 
+    if (APPLE)
+      add_custom_command(TARGET ${BASE_NAME}_tests
+        POST_BUILD
+        COMMAND ${CMAKE_INSTALL_NAME_TOOL} -add_rpath "${openstudio_ROOT_DIR}/lib/" $<TARGET_FILE:${BASE_NAME}_tests>
+        COMMAND ${CMAKE_INSTALL_NAME_TOOL} -add_rpath "${QT_INSTALL_DIR}/lib/" $<TARGET_FILE:${BASE_NAME}_tests>)
+    endif()
+
```

Comment:
shouldn't we do something like this?

```cmake
  set_target_properties(${BASE_NAME}_tests PROPERTIES
    BUILD_RPATH <...>
    INSTALL_RPATH <...>
  )
```

Reply:
@macumber up to you if you want to make that change (feels more "Cmake-ish" to me, but if it works like this I'm fine not addressing). Merge away when you are content and let's issue an RC1

Reply:
@macumber this is because conan(.cmake) will set `CMAKE_SKIP_RPATH=True` 

We'd need KEEP_RPATHS here: https://github.com/openstudiocoalition/OpenStudioApplication/blob/589b2bf4ef0ffdca3a9ab7fcc076c02e927c5012/ConanInstall.cmake#L195

This is what I did on the OS SDK side: https://github.com/NREL/OpenStudio/commit/091c148fadbd6464d1f62d288525165cf3827a71

https://cmake.org/cmake/help/latest/variable/CMAKE_SKIP_RPATH.html

> If true, do not add run time path information.

> If this is set to TRUE, then the rpath information is not added to compiled executables. The default is to add rpath information if the platform supports it. This allows for easy running from the build tree. To omit RPATH in the install step, but not the build step, use [CMAKE_SKIP_INSTALL_RPATH](https://cmake.org/cmake/help/latest/variable/CMAKE_SKIP_INSTALL_RPATH.html#variable:CMAKE_SKIP_INSTALL_RPATH) instead.